src/main/java/ui/UserPresenter.java

```diff
@@ -0,0 +1,38 @@
+package ui;
+
+import host_room.HostRoomController;
+import host_room.HostRoomInteractor;
+import host_room.HostRoomPresenter;
+import database.RoomDBGateway;
+import signup_and_login.UserOutputBoundary;
+
+import javax.swing.*;
+
+public class UserPresenter implements UserOutputBoundary {
+    private RoomDBGateway roomDBAccess;
+    private Integer currUserID;
+    public UserPresenter(RoomDBGateway roomDBAccess, Integer currUserID) {
+        this.roomDBAccess = roomDBAccess;
+        this.currUserID = currUserID;
+    }
+    /**
+     * Displays the JoinOrHostView passing in the specified user ID.
+     * @param userID the ID of the user
+     */
+    @Override
+    public void prepareJoinOrHostView(int userID) {
+        HostRoomPresenter hostRoomPresenter = new HostRoomPresenter();
+        HostRoomInteractor hostRoomInteractor = new HostRoomInteractor(roomDBAccess, hostRoomPresenter);
+        HostRoomController hostRoomController = new HostRoomController(hostRoomInteractor);
+        JoinOrHostView joinOrHostView = new JoinOrHostView(hostRoomController, currUserID);
```

Comment:
I'm not sure this is very CA. Is it possible to initialize these in `Main` and make prepareView() a static method so that we can call it from here? If it's not we should definitely figure out a solution for this - maybe we ask TA on Thursday.